contracts/NonFungibleToken-v2.cdc

```diff
@@ -260,24 +260,24 @@ access(all) contract NonFungibleToken {
 
         /// deposit takes a NFT and adds it to the collections dictionary
         /// and adds the ID to the id array
-        access(all) fun deposit(token: @{NonFungibleToken.NFT}) 
-        // {
-        //     pre {
-        //         // We emit the deposit event in the `Collection` interface
-        //         // because the `Collection` interface is almost always the final destination
-        //         // of tokens and deposit emissions from custom receivers could be confusing
-        //         // and hard to reconcile to event listeners
-        //         //NonFungibleToken.emitNFTDeposit(id: token.getID(), uuid: token.uuid, to: self.owner?.address, type: token.getType().identifier)
-        //     }
-        // }
+        access(all) fun deposit(token: @{NonFungibleToken.NFT}) {
+            pre {
+                // We emit the deposit event in the `Collection` interface
+                // because the `Collection` interface is almost always the final destination
+                // of tokens and deposit emissions from custom receivers could be confusing
+                // and hard to reconcile to event listeners
+                NonFungibleToken.emitNFTDeposit(id: token.getID(), uuid: token.uuid, to: self.owner?.address, type: token.getType().identifier)
+            }
+        }
 
         /// Function for a direct transfer instead of having to do a deposit and withdrawal
         /// This can and should return false if the transfer doesn't succeed and true if it does succeed
         ///
         access(Withdrawable) fun transfer(id: UInt64, receiver: Capability<&{NonFungibleToken.Receiver}>): Bool {
             pre {
                 receiver.check(): "Could not borrow a reference to the NFT receiver"
-                //NonFungibleToken.emitNFTTransfer(id: id, uuid: self.borrowNFTSafe(id: id)?.uuid, from: self.owner?.address, to: receiver.borrow()?.owner?.address, type: self.borrowNFT(id).getType().identifier)
+                self.getIDs().contains(id): "The collection does not contain the specified ID"
```

Comment:
Good point. FWIW I think it's best to remove it to keep the interface flexible, but worth bringing up in the open house this week

Reply:
If we want to proceed here, the problem then is in the `NonFungibleToken.emitNFTTransfer(..., type: self.borrowNFT(id).getType().identifier)`. We'll need to determine if we can emit `String?` otherwise the `self.getIDs().contains(id)` pre-condition remains necessary.